### PR TITLE
[Feature] according to es2.0 spec, you must use clamp-to-edge as  tex…

### PR DESCRIPTION
**Pull request by Mee-gu**
…ture wrap mode, and no mipmapping for non-power-of-two texture

### CODE REVIEW

File: cocos/renderer/backend/Types.h [hunk 1]
```diff
@@ -164,8 +164,8 @@ struct SamplerDescriptor
     SamplerFilter magFilter = SamplerFilter::LINEAR;
     SamplerFilter minFilter = SamplerFilter::LINEAR;
     SamplerFilter mipmapFilter = SamplerFilter::LINEAR;
-    SamplerAddressMode sAddressMode = SamplerAddressMode::REPEAT;
-    SamplerAddressMode tAddressMode = SamplerAddressMode::REPEAT;
```
Comment (minggo) [inline cocos/renderer/backend/Types.h:168]: It is ok to change default value, but we should do some protection of the usage in Texture.

Comment (Mee-gu) [reply cocos/renderer/backend/Types.h:168]: got it.

File: cocos/renderer/backend/opengl/TextureGL.h [hunk 2]
```diff
@@ -5,6 +5,8 @@
 
 CC_BACKEND_BEGIN
 
+#define ISPOW2(n) (((n)&(n-1)) == 0)
```
Comment (minggo) [inline cocos/renderer/backend/opengl/TextureGL.h]: It is better to define it in .cpp file.

File: cocos/renderer/backend/opengl/TextureGL.cpp [hunk 3]
```diff
@@ -55,9 +64,12 @@ namespace
         }
     }
     
-    GLint toGLAddressMode(SamplerAddressMode addressMode)
+    GLint toGLAddressMode(SamplerAddressMode addressMode, bool bPow2)
     {
-        GLint ret = GL_REPEAT;
+        GLint ret = GL_CLAMP_TO_EDGE;
+        if(!bPow2)
+            return ret;
```
Comment (minggo) [inline cocos/renderer/backend/opengl/TextureGL.cpp:75]: It is better to output a log to say that the address mode is changed to `GL_GLAMP_TO_EDGE`.

File: cocos/renderer/backend/opengl/TextureGL.cpp [hunk 4]
```diff
@@ -22,10 +22,19 @@ namespace
...
     }
     
-    GLint toGLMinFilter(SamplerFilter minFilter, SamplerFilter mipmapFilter, bool mipmapEnabled)
+    GLint toGLMinFilter(SamplerFilter minFilter, SamplerFilter mipmapFilter, bool mipmapEnabled, bool bPow2)
     {
         if (mipmapEnabled)
         {
+            if(!bPow2)
+            {
+                cocos2d::log("Disable mipmap since non-power-of-two texture occur in %s %s %d", __FILE__, __FUNCTION__, __LINE__);
```
Comment (minggo) [inline cocos/renderer/backend/opengl/TextureGL.cpp]: It is better to say that what filter is used instead.

File: cocos/renderer/backend/opengl/TextureGL.cpp [hunk 5]
```diff
@@ -80,13 +92,13 @@ TextureGL::TextureGL(const TextureDescriptor& descriptor) : Texture(descriptor)
 {
     glGenTextures(1, &_texture);
     toGLTypes();
-    
+    bool bPow2 = ISPOW2(_width) && ISPOW2(_height);
```
Comment (minggo) [inline cocos/renderer/backend/opengl/TextureGL.cpp]: We don't use Hungary style, so it is better to change to `isPow2`.

Comment (Mee-gu) [reply cocos/renderer/backend/opengl/TextureGL.cpp]: got it.

File: cocos/renderer/backend/opengl/TextureGL.cpp (hunk 4 above)
Comment (Mee-gu) [reply cocos/renderer/backend/opengl/TextureGL.cpp]: Done!

File: cocos/renderer/backend/opengl/TextureGL.cpp (hunk 3 above)
Comment (Mee-gu) [reply cocos/renderer/backend/opengl/TextureGL.cpp:75]: Done!

File: cocos/renderer/backend/opengl/TextureGL.h (hunk 2 above)
Comment (Mee-gu) [reply cocos/renderer/backend/opengl/TextureGL.h]: Done!

**Pull request merged**